lib/view/primary_button.dart

```diff
@@ -139,3 +140,55 @@ class OutlineButton extends StatelessWidget {
     );
   }
 }
+
+class PrimaryAsyncButton extends StatefulWidget {
+  final Function()? onTap;
+  final Color? color;
+  final String? text;
+  final double? width;
+  final bool enabled;
+  final String? processingText;
+
+  const PrimaryAsyncButton(
+      {Key? key,
+      this.onTap,
+      this.color,
+      this.text,
+      this.width,
+      this.enabled = true,
+      this.processingText})
+      : super(key: key);
+
+  @override
+  State<PrimaryAsyncButton> createState() => _PrimaryAsyncButtonState();
+}
+
+class _PrimaryAsyncButtonState extends State<PrimaryAsyncButton> {
+  bool _isProcessing = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return PrimaryButton(
+      onTap: () {
+        withDebounce(
+          () async {
+            setState(() {
```

Comment:
we should add a check of `_isPrecessing == true` for case when time of function call is more than debounce duration

Reply:
why? the loading indicator show until atleast the function finished